Sources/HTTPParser/http_parser.swift

```diff
@@ -61,20 +173,44 @@ private let HTTP_MAX_HEADER_SIZE = 80*1024
  * many times for each string. E.G. you might get 10 callbacks for "on_url"
  * each providing just a few characters more data.
  */
-public protocol http_parser_delegate {
-  func on_message_begin() -> Int
-  func on_url(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_status(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_header_field(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_header_value(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_headers_complete() -> Int
-  func on_body(at: UnsafePointer<UInt8>, length: Int) -> Int
-  func on_message_complete() -> Int
+public struct http_parser_delegate {
```

Comment:
Using escaping closures instead of a protocol was way slower in my tests.

Unrelated, missing in the original too: the callbacks need the parser as the first argument (like in the original) to provide the necessary context (w/o requiring capturing closures).